imagedephi/rules.py

```diff
@@ -9,10 +9,20 @@ class FileFormat(Enum):
     SVS = "svs"
 
 
+expected_type_map: dict[str, list[Type[Any]]] = {
+    "integer": [int],
+    "number": [int, float],
+    "text": [str],
+    "rational": [int],
+}
+
+RedactionOperation: TypeAlias = Literal["keep", "delete", "replace"]
```

Comment:
Our rule definition breaks down a bit with this change. Previously the rule action described what would be done to a particular piece of metadata upon redaction, but that doesn't apply to the new `check_type` rules. The actual thing that will be done is determined at runtime, and is represented by a new type, `RedactionOperation`.